test/common/database.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const Database = require('../../src/server/database/database-abstraction.js')
+const assert = require('assert')
+
+describe('Database', () => {
+  it('should create a table', (done) => {
+    let db = Database.create()
+
+    db.connect().then(() => {
+      db.table('test').then(() => { done() })
+    })
+  })
+})
+
+describe('Database', function () {
+  it('should put, get, update and remove a document properly', (done) => {
+    let db = Database.create()
```

Comment:
I know why there are together in one test, but I have the tendency to split those kind of tests into three:
- 1 insert (on empty and non-empty DB)
- 1 update on something that is already in the database and which I know (using test setup)
- 1 delete on something that is already in the db too (using test setup)
I find it adds granularity to the tests (and I like to test only one thing in a unit test, if possible).